test/functional/lua/buffer_updates_spec.lua

```diff
@@ -349,7 +349,12 @@ describe('lua: nvim_buf_attach on_bytes', function()
       end
 
       local text = meths.buf_get_lines(0, 0, -1, true)
-      local bytes = table.concat(text, '\n') .. '\n'
+      local bytes = table.concat(text, '\n')
```

Comment:
this is not correct. even an empty (loaded) buffer still has a final line ending.

Reply:
Ok I see. So it either a special case for edit operations which lead to empty buffer or a bug =) Thanks, will investigate it.